.github/workflows/check_issue_label.yml

```diff
@@ -0,0 +1,96 @@
+            const owner = context.repo.owner;
+            const repo = context.repo.repo;
+            console.log("Label Name:", labelName);
+            console.log("Username:", username);
+            console.log("Issue Number:", issueNumber);
+            const isValidLabelName = /^[\w\s-:]+$/.test(labelName); 
+            if (!isValidLabelName) {
+              console.error("Invalid label name:", labelName);
+              return;
+            }
+            if (!Number.isInteger(issueNumber) || issueNumber <= 0) {
+              console.error("Invalid issue number:", issueNumber);
+              return;
+            }
+            const isValidUsername = /^[a-zA-Z0-9-]+$/.test(username);
+            if (!isValidUsername) {
+              console.error("Invalid username:", username);
+              return;
+            }
+            // Load allowed users from userwhitelist.json
+            let allowedUsers = [];
+            try {
+              const userWhitelistPath = path.join(process.env.GITHUB_WORKSPACE, 'userWhitelist.json');
+              const userWhitelist = JSON.parse(fs.readFileSync(userWhitelistPath, 'utf8'));
+              allowedUsers = userWhitelist.goodFirstIssue || [];
+            } catch (error) {
+              console.error("Error loading or parsing userWhitelist.json:", error);
+              return; // Exit if the file is missing or invalid
+            }
+            const goodFirstIssueUsers = new Set(allowedUsers);
+            const prLabels = new Set(["dependencies", "stale"]);
+            async function removeLabelAndComment(label, comment) {
+              console.log(`Removing label: ${label}`);
+              await github.rest.issues.removeLabel({
+                owner,
+                repo,
+                issue_number: issueNumber,
+                name: label
+              }).catch(err => console.error(`Error removing label: ${err.message}`));
+              console.log(`Creating comment: ${comment}`);
+              const escapeHTML = (str) =>
+                str.replace(/&/g, "&amp;").replace(/</g, "&lt;").replace(/>/g, "&gt;").replace(/"/g, "&quot;");
+              const sanitizedComment = escapeHTML(comment);
+              await github.rest.issues.createComment({
+                owner,
+                repo,
+                issue_number: issueNumber,
+                body: sanitizedComment
+              }).catch(err => console.error(`Error creating comment: ${err.message}`));
+            }
+            // only allowlisted users can add "good first issue" label
+            if (labelName === "good first issue") {
+              console.log("Checking if user is allowed...");
+              if (!goodFirstIssueUsers.has(username)) {
+                await removeLabelAndComment(
+                  labelName,
+                  `Hi @${username}, only users from the 'goodFirstIssue' team can add the 'good first issue' label. Please reach out to them if needed. Thanks!`
```

Comment:
please reference the https://github.com/orgs/oppia/teams/oppia-good-first-issue-labelers team so devs can find who is on this list